src/desktops/dskSelectMap.cpp

```diff
@@ -91,6 +96,10 @@ dskSelectMap::dskSelectMap(const CreateServerInfo& csi)
     AddTextButton(4, 590, 530, 200, 22, TC_GREEN2, _("Load game..."), NormalFont);
     // "Weiter"
     AddTextButton(5, 590, 560, 200, 22, TC_GREEN2, _("Continue"), NormalFont);
+    // random map generation
+    AddTextButton(6, 380, 530, 150, 22, TC_GREEN2, _("Create Random"), NormalFont);
+    // random map settings
+    AddTextButton(7, 540, 530, 40, 22, TC_GREEN2, _("..."), NormalFont);
```

Comment:
Long-term it would be nice to have the map generated once the game starts (so people don't see the mini-map before). That was my initial thought, but I couldn't figure out which pieces of code to touch for this (seem to be quite time consuming to get that far), so I started with this "light-weight" approach which can be implemented most independently existing code.